components/camel-pulsar/src/main/java/org.apache.camel.component.pulsar/PulsarProducer.java

```diff
@@ -46,65 +47,37 @@ public static PulsarProducer create(final PulsarEndpoint pulsarEndpoint) {
-        synchronized (this) {
-            if (!producers.isEmpty()) {
-                final Message message = exchange.getIn();
-                //TODO is peek the correct method to use here - look at the head of the q but do not remove?
-                final Producer<byte[]> producer = producers.peek();
-
-                try {
-                    byte[] body = exchange.getContext().getTypeConverter()
-                        .mandatoryConvertTo(byte[].class, exchange, message.getBody());
-                    producer.send(body);
-
-                } catch (NoTypeConversionAvailableException | TypeConversionException exception) {
-                    LOGGER.debug("An error occurred while serializing to byte array, fall using fall back strategy :: {}", exception);
-
-                    byte[] body = PulsarMessageUtils.serialize(message.getBody());
-
-                    producer.send(body);
-                }
-            } else {
-                LOGGER.error("No producer associated with endpoint [{}]", pulsarEndpoint.getEndpointUri());
-            }
-        }
-    }
+        final Message message = exchange.getIn();
+        final String topic = pulsarEndpoint.getTopic();
+        final String producerName = topic + "-" + Thread.currentThread().getId();
 
-    // TODO why is this synchronized but other methods are not?
-    // what are we protecting from concurrency issues - is it the producers collection?
-    // if it is the producers collection, would it be better/safer to wrap that in Collections.synchronizedCollection
-    @Override
-    protected synchronized void doStart() throws Exception {
-        super.doStart();
+        final Map properties = message.getHeader(PulsarMessageHeaders.PROPERTIES, Map.class);
 
-        producers = stopProducer(producers);
-        producers.add(createProducer(pulsarEndpoint));
-    }
+        final ProducerBuilder<byte[]> producerBuilder = pulsarEndpoint
+            .getPulsarClient()
+            .newProducer()
+            .producerName(producerName)
+            .topic(topic);
 
+        if (properties != null && !properties.isEmpty()) {
+            producerBuilder.properties(properties);
+        }
 
-    @Override
-    protected void doStop() throws Exception {
-        super.doStop();
-        producers = stopProducer(producers);
-    }
+        final Producer<byte[]> producer = producerBuilder.create();
 
-    @Override
-    protected void doSuspend() throws Exception {
-        super.doSuspend();
+        try {
+            byte[] body = exchange.getContext().getTypeConverter()
+                .mandatoryConvertTo(byte[].class, exchange, message.getBody());
+            producer.send(body);
 
-        producers = stopProducer(producers);
-    }
+        } catch (NoTypeConversionAvailableException | TypeConversionException exception) {
+            LOGGER.warn("An error occurred while serializing to byte array, fall using fall back strategy :: {}", exception);
```

Comment:
I've just changed my mind about this level of logging from debug to warn. The reason I've changed my mind is that it should trigger the user of this project to add a TypeConverter into their camel context due to this message spamming their logs.